models/user.go

```diff
@@ -263,11 +263,11 @@ func (u *User) RelAvatarLink() string {
 	return setting.GravatarSource + u.Avatar
 }
 
-// AvatarLink returns user gravatar link.
+// AvatarLink returns user avatar link.
 func (u *User) AvatarLink() string {
 	link := u.RelAvatarLink()
 	if link[0] == '/' && link[1] != '/' {
-		return setting.AppSubUrl + link
+		return strings.TrimSuffix(setting.AppUrl, "/") + link
```

Comment:
I just remembered the rationale for my approach:
if AppUrl is in the form "https://try.gogs.io"
your version would result in a bogus url
(ie: "https://try.gogs.ioavatar/3")

So my code is trimming the AppUrl instead, knowing that
the avatar link starts with a slash (as per prior check).